tests/nn/moe/test_moelayer.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+import torch
+
+from fairscale.nn import MOELayer, Top2Gate
+
+skip_if_no_cuda = pytest.mark.skipif(not torch.cuda.is_available(), reason="cuda required")
+
+
+def test_create():
+    model_dim = 8
+    num_experts = 4
+    gate = Top2Gate(model_dim, num_experts)
+    expert = torch.nn.Linear(model_dim, model_dim)
```

Comment:
I am a bit confused on the MOELayer API here. Will it only take a single expert or a network of experts? It seems like it is the former. Does that mean that some how multiple instances of MOELayer will need to be created and share the gate network? That's why the class variable above where I asked a question? Maybe there is an assumption somewhere that I am not capturing? Sorry about the dumb question.

Reply:
Yes. Multiple instances will be created. Each device will instantiate its own expert.